lib/share_progress/button.rb

```diff
@@ -32,8 +47,42 @@ def endpoint(method=nil)
       end
 
       def filter_keys(params, acceptable)
```

Comment:
the problem though isn't that `params` aren't passed, but rather that they're passed as nil because I'm passing `params[:advanced_options]` as `params`, and that's nil if the key doesn't exist. could be tidied up a different way maybe though.